src/bam/call_consensus_reads/pipeline.rs

```diff
@@ -26,13 +27,13 @@ type RecordIDS = Vec<RecordID>;
 type RecordID = Vec<u8>;
 
 impl CallConsensusRead {
-    pub fn call_consensus_reads(&mut self) -> Result<(), Box<dyn Error>> {
+    pub fn call_consensus_reads(&mut self) -> errors::Result<()> {
         let mut group_end_idx: BTreeMap<Position, GroupIDs> = BTreeMap::new();
         let mut duplicate_groups: HashMap<GroupID, RecordIDS> = HashMap::new();
         let mut record_storage: HashMap<RecordID, RecordStorage> = HashMap::new();
 
         for (i, result) in self.bam_reader.records().into_iter().enumerate() {
-            let mut record = result?;
+            let mut record = result.context(errors::BamReadError { record: None })?;
```

Comment:
Why is there at all a record in BamReadError?

Reply:
Maybe rather add `i` to the error, so that people can see which record caused the error?

Reply:
You are right. The record does not make sense at this place. I'll replace it by its index.